tensorflow/java/src/main/native/saved_model_bundle_jni.cc

```diff
@@ -0,0 +1,101 @@
+/* Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include <limits>
+#include <memory>
+
+#include "tensorflow/c/c_api.h"
+#include "tensorflow/java/src/main/native/exception_jni.h"
+#include "tensorflow/java/src/main/native/saved_model_bundle_jni.h"
+
+namespace {
+
+}  // namespace
+JNIEXPORT jobject JNICALL Java_org_tensorflow_SavedModelBundle_load(JNIEnv * env,
+                                                                    jclass clazz,
+                                                                    jstring export_dir,
+                                                                    jobjectArray tags) {
+
+  TF_Status* status = TF_NewStatus();
+  jobject bundle = nullptr;
+
+  // allocate parameters for TF_LoadSessionFromSavedModel
+  TF_SessionOptions* opts = TF_NewSessionOptions();
+  const TF_Buffer* crun_options = nullptr;
+  const char* cexport_dir = env->GetStringUTFChars(export_dir, nullptr);
+  std::unique_ptr<const char* []> tags_ptrs;
+  size_t tags_len = env->GetArrayLength(tags);
+  tags_ptrs.reset(new const char*[tags_len]);
+  for (size_t i = 0; i < tags_len; ++i) {
+    jstring tag = static_cast<jstring>(env->GetObjectArrayElement(tags, i));
+    tags_ptrs[i] = env->GetStringUTFChars(tag, nullptr);
+    env->DeleteLocalRef(tag);
+  }
+
+  // load the session
+  TF_Graph* graph = TF_NewGraph();
+  TF_Buffer* metagraph_def = TF_NewBuffer();
+  TF_Session* session = TF_LoadSessionFromSavedModel(opts, crun_options, cexport_dir,
+                                                     tags_ptrs.get(), tags_len, graph,
+                                                     metagraph_def, status);
+
+  // release the parameters
+  TF_DeleteSessionOptions(opts);
+  env->ReleaseStringUTFChars(export_dir, cexport_dir);
+  for (size_t i = 0; i < tags_len; ++i) {
+    jstring tag = static_cast<jstring>(env->GetObjectArrayElement(tags, i));
+    env->ReleaseStringUTFChars(tag, tags_ptrs[i]);
+    env->DeleteLocalRef(tag);
+  }
+
+  // handle the result
+  if (throwExceptionIfNotOK(env, status)) {
+    // sizeof(jsize) is less than sizeof(size_t) on some platforms.
+    if (metagraph_def->length > std::numeric_limits<jint>::max()) {
```

Comment:
I copied this snippet from `graph_jni.cc` so I'd prefer you follow up on that, OK?